fe/lib/components/LocationSelectMap/LocationSelectMap.tsx

```diff
@@ -0,0 +1,191 @@
+    tablet: number;
+    mobile: number;
+  };
+}
+
+export const LocationSelectMap = ({
+  debounceDelay = 250,
+  schemeId,
+  onLocationSelected,
+  initialLocation,
+  client,
+  mapHeight = {
+    wide: DEFAULT_MAP_HEIGHT,
+    desktop: DEFAULT_MAP_HEIGHT,
+    tablet: 200,
+    mobile: 200,
+  },
+}: LocationSelectMapProps) => {
+  const [latLong, setLatLong] = useState(initialLocation);
+  const [showSuggestions, setShowSuggestions] = useState(false);
+
+  const responsiveValue = useResponsiveValue();
+  const color = useColor();
+  const [debouncedLatLong] = useDebounce(latLong, debounceDelay);
+
+  const { data, previousData } = useQuery<NearbyLocationsQuery>(
+    NEAREST_LOCATIONS,
+    {
+      variables: {
+        geoLocation: {
+          latitude: debouncedLatLong[0],
+          longitude: debouncedLatLong[1],
+        },
+        schemeId,
+      },
+      client,
+    },
+  );
+
+  const nearestLocations =
+    data?.nearestLocations ?? previousData?.nearestLocations;
+
+  const isDesktopOrAbove = responsiveValue({
+    mobile: false,
+    desktop: true,
+  });
+
+  const height = responsiveValue(mapHeight) ?? DEFAULT_MAP_HEIGHT;
+
+  const NearestLocationsCards = () => (
+    <Box borderRadius="standard" background="surface">
+      <Stack dividers space="none">
+        <Box
+          paddingX="medium"
+          paddingY={{ desktop: 'xsmall', tablet: 'small', mobile: 'small' }}
+        >
+          <Columns space="small" alignY="center">
+            <Column>
+              <Text weight="strong" size="small">
+                {`Locations near ${formatPoint(latLong)}`}
+              </Text>
+            </Column>
+            {isDesktopOrAbove ? (
+              <Column width="content">
+                <ButtonIcon
+                  icon={<IconClear />}
+                  tone="secondary"
+                  id="clear-nearest-suggestions-button"
+                  label="Clear Suggestions"
+                  onClick={() => setShowSuggestions(false)}
+                />
+              </Column>
+            ) : null}
+          </Columns>
+        </Box>
+        {nearestLocations?.map((location) => (
+          <Box
+            className={styles.card}
+            cursor="pointer"
+            padding="medium"
+            key={location.id.value}
+            onClick={() => onLocationSelected(location)}
+          >
+            <Stack space="small">
+              <Text size="small">{location.contextualName}</Text>
+              <InlineCode>{location.id.value}</InlineCode>
+            </Stack>
+          </Box>
+        ))}
+      </Stack>
+    </Box>
+  );
+
+  const NoLocationsNotice = () => (
+    <Alert
+      tone="info"
+      onClose={() => setShowSuggestions(false)}
+      closeLabel="Close"
+    >
+      <Text>No locations found.</Text>
+    </Alert>
+  );
+
+  const hasNearestLocations = nearestLocations && nearestLocations.length > 0;
+  const suggestionsOffset: [number, number] = [-40, 130];
```

Comment:
Noob question, is there any significance to these magic numbers?

Reply:
These control where the suggestions will be overlayed in relation to a point. An offset of `[0,0]`, i.e, no offset, would position the suggestions with the marker like this:
![Screen Shot 2022-10-17 at 5 08 44 pm](https://user-images.githubusercontent.com/30482449/196101421-eb81ea07-f6d8-4ec7-b6b2-6fdb63d436bf.png)

Specifying `[-40, 130] `moves the suggestions 40 pixels to the right of the marker and 130 pixels up from the marker.

Reply:
I'll add a comment on the offset so this logic is clear

Reply:
All good. What I meant to ask is how we decided on `-40` and `130` specifically, and whether they can be related to other numbers in the codebase (e.g. the height of the marker icon) or maybe even Braid's [spacing scale](https://seek-oss.github.io/braid-design-system/foundations/layout#spacing).

Reply:
Ah ok gotcha. Those unfortunately don't relate to anything else and I just picked them by sight. Will try see if I can get them to relate to either of those things 👍

Reply:
I've gone with Braid's [useSpace](https://seek-oss.github.io/braid-design-system/components/useSpace) hook for the X offset, but after exploring options for getting a dynamic/relative Y offset I think the only option would be doing funky stuff with refs, and that feels a bit clunky. So I'm thinking I will just keep the `130` for now